pkg/load/agents/configAgent_test.go

```diff
@@ -175,3 +175,183 @@ func TestBuildIndexes(t *testing.T) {
 		})
 	}
 }
+
+func TestConfigAgent_GetMatchingConfig(t *testing.T) {
+	var testCases = []struct {
+		name        string
+		input       load.ByOrgRepo
+		meta        api.Metadata
+		expected    api.ReleaseBuildConfiguration
+		expectedErr bool
+	}{
+		{
+			name:  "no configs in org fails",
+			input: load.ByOrgRepo{},
+			meta: api.Metadata{
+				Org:    "org",
+				Repo:   "repo",
+				Branch: "branch",
+			},
+			expectedErr: true,
+		},
+		{
+			name: "no configs in repo fails",
+			input: load.ByOrgRepo{
+				"org": map[string][]api.ReleaseBuildConfiguration{},
+			},
+			meta: api.Metadata{
+				Org:    "org",
+				Repo:   "repo",
+				Branch: "branch",
+			},
+			expectedErr: true,
+		},
+		{
+			name: "no configs for variant fails",
+			input: load.ByOrgRepo{
+				"org": map[string][]api.ReleaseBuildConfiguration{
+					"repo": {{Metadata: api.Metadata{
+						Org:    "org",
+						Repo:   "repo",
+						Branch: "branch",
+					}}},
+				},
+			},
+			meta: api.Metadata{
+				Org:     "org",
+				Repo:    "repo",
+				Branch:  "branch",
+				Variant: "variant",
+			},
+			expectedErr: true,
+		},
+		{
+			name: "literal match returns it",
+			input: load.ByOrgRepo{
+				"org": map[string][]api.ReleaseBuildConfiguration{
+					"repo": {{Metadata: api.Metadata{
+						Org:    "org",
+						Repo:   "repo",
+						Branch: "branch",
+					}}},
+				},
+			},
+			meta: api.Metadata{
+				Org:    "org",
+				Repo:   "repo",
+				Branch: "branch",
+			},
+			expected: api.ReleaseBuildConfiguration{Metadata: api.Metadata{
+				Org:    "org",
+				Repo:   "repo",
+				Branch: "branch",
+			}},
+			expectedErr: false,
+		},
+		{
+			name: "regex match on branch returns it",
+			input: load.ByOrgRepo{
+				"org": map[string][]api.ReleaseBuildConfiguration{
+					"repo": {{Metadata: api.Metadata{
+						Org:    "org",
+						Repo:   "repo",
+						Branch: "branch",
```

Comment:
This testcase confuses me, why does it pass, shouldn't the `Branch` be a `branch.*` to have it pass?

Reply:
Regex doesn't need to match the full input!